### ISSUE TITLE
Make "show" functionality in codename on /lookup page clearer

### ISSUE THREAD

**Issue by redshiftzero** [freedomofpress/securedrop#3586]
## Description

The first time users view the source interface, they are shown their codename on the lookup page, in case they forget to note the codename down when first shown on the generate page. It appears, but not by default here:

![screen shot 2018-06-26 at 4 19 33 pm](https://user-images.githubusercontent.com/7832803/41944513-fe59d346-795c-11e8-9e45-7525ddf328c7.png)

Some users find this non-obvious (see note below) - perhaps due to the distance of the "Show" button off to the right. Any ideas for how to make this clearer? cc @huertanix @harrislapiroff @ei8fdb

## User Research Evidence

A user submitting through the source interface for the first time in a while says: "i forgot to copy my code name down after the first page. when i looked to the bottom where it says "Remember, your codename is:" I was kind of confused that it didn't show it to me, and the "show" button was off to the side and took me a second [..snip..] to figure out what was going on"

**Comment by harrislapiroff** [comment]
Here's an off the cuff first idea

![image](https://user-images.githubusercontent.com/296631/41981696-896ba488-79f7-11e8-8eb1-514f718e60f4.png)

**Comment by huertanix** [comment]
^ I like Harris's idea. Reminds me of the Mastodo content warning's click-to-view design. 

<img width="714" alt="wednesday" src="https://user-images.githubusercontent.com/72914/42109739-b31a4e40-7bac-11e8-943e-3371ffa132e3.png">

I would only make it look more clickable (SD styleguide for clickable things may come into play here). Maybe with clickable styling the wording can just be "Show codename".

**Comment by ei8fdb** [comment]
@redshiftzero this makes a lot of sense. I agree with your comment about the distance between the control and the UI area it controls - you've essentially hit on [Fitts's Law](https://www.interaction-design.org/literature/article/fitts-s-law-the-importance-of-size-and-distance-in-ui-design) (a law used for predictive modelling of human movement): which says It's faster to hit large targets closer to you than it is to hit smaller targets further away from you. (There is also a connection with the size of the interactive element)

@harrislapiroff's off the cuff idea makes a lot of sense - the user's attention will be drawn towards the words - I'd expect a user to try clicking the text. This would reveal their codename.

* Riff 1
I'd go a bit further and actually make the whole element clickable (back to Fitts's law). If the objective is to make the user click to reveal their codename, this will increase the possibility of the user clicking it, in terms of accuracy.

![securedrop___protecting_journalists_and_sources](https://user-images.githubusercontent.com/1899506/42117691-03ded5dc-7bf6-11e8-8917-39a6bb57eaeb.png)

* Riff 2
I have no evidence, but it could also be the case that the user is not aware of the significance of the "unlabelled" element at the bottom of the page.

This could be solved by giving the codename box a label.

![securedrop___protecting_journalists_and_sources](https://user-images.githubusercontent.com/1899506/42117664-ccf57c24-7bf5-11e8-9e39-08fd5a628c6e.png)

Comments?

--

I will do some usability testing of the current design (4-5 people) and the proposed designs to see results.

**Comment by redshiftzero** [comment]
This is great feedback @harrislapiroff @huertanix @ei8fdb! The large clickable area makes a lot of sense to me and I think it would be a solid improvement to the UI. I suspect including the "Remember your codename is" or "Your codename" will help users understand what the clickable area is for, but I defer to the results of the usability testing :muscle:

**Comment by huertanix** [comment]
@ei8fdb I agree with all of this. Labeling also helps in training since now we'll be able to simply say "scroll down to Your Codename" rather than ask trainees to look for a gray box with long tiny text. 💯

**Comment by ninavizz** [comment]
Has this been resolved (like—I think what Bernard recommended is in the current UI?) @redshiftzero?

**Comment by ninavizz** [comment]
(I'd still love to see research results if that ever happened, @ei8fdb!)

**Comment by eloquence** [comment]
It's worth noting that this UI was changed a while ago and is now at the top of the page:
![showcodename](https://user-images.githubusercontent.com/213636/107432701-32f27800-6add-11eb-9559-6eb62e32e2ca.png)
However, alternative approaches for the widget design are still worth considering, so leaving this issue open.

See also the related observation in #5239 that the scrolling behavior could be more harmful than useful at this point.

**Comment by Gomesz785** [comment]
can anyone note the file that contains the UI coding for following screen?
![image](https://user-images.githubusercontent.com/87658122/126346479-fed29995-0ea8-4c3f-9750-d22aa3dc19c0.png)

**Comment by Gomesz785** [comment]
> can anyone note the file that contains the UI coding for following screen?
> ![image](https://user-images.githubusercontent.com/87658122/126346479-fed29995-0ea8-4c3f-9750-d22aa3dc19c0.png)

so i can improve it...

**Comment by zenmonkeykstop** [comment]
Hi @Gomesz785 - this UI element is in: https://github.com/freedomofpress/securedrop/blob/develop/securedrop/source_templates/lookup.html

UX changes in the Source Interface do tend to be pretty carefully considered, so if you do want to try to make any changes, please update this issue with an implementation plan first before getting too deep into it.

**Comment by ninavizz** [comment]
Hi @Gomesz785 Thank you for your interest in contributing! That UI element has actually been moved to the top of the screen, and a number of other things are also different about the overall UX, today, from how the UI looks in the screenshots above. You can poke around what the current experience looks like, here: http://demo.securedrop.org/

I'm a UX design and research contributor (yes data, yes pixels, no code), and have been casually working on this problem for a while. You can see some of what I've worked on in [my most current prototype](https://projects.invisionapp.com/share/ZH115V9DERP2#/screens)(made using a visual prototyping tool called InVision). Feel free to also browse our wiki and Issues that track UX work for SecureDrop, here: https://github.com/freedomofpress/securedrop-ux.

Our UX Meetings are every-other-Thursday, and [this Thursday we will be discussing](https://github.com/freedomofpress/securedrop-ux/wiki/Meetings) stuff relevant to the Source UI. If you'd like to attend for that design review to get a sense for how we work, we'd love to have you!

**Comment by legoktm** [comment]
The codename widget was overhauled in #6315, it now looks like:

| Hidden | Shown |
| --- | --- |
| ![Screenshot 2022-09-30 at 20-13-42 SecureDrop Protecting Journalists and Sources](https://user-images.githubusercontent.com/81392/193373781-bec238e6-2b75-4fa6-b380-bf67e1caa9e0.png) | ![Screenshot 2022-09-30 at 20-13-49 SecureDrop Protecting Journalists and Sources](https://user-images.githubusercontent.com/81392/193373795-cd23520d-7e30-444e-9436-ae2eca8f4062.png) |

@eaon, @tina-ux: I think this can be closed now?

**Comment by zenmonkeykstop** [comment]
Closing as per above.
